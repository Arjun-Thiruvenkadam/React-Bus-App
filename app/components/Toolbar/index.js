/**
 *
 * Toolbar
 *
 */

import React from 'react';
import Logo from '../Logo/index';
import NavigationItems from '../../containers/Navigation/index';
import './toolbar.scss';

const Toolbar = () => (
  <header className="Toolbar">
    <Logo />
    <nav>
      <NavigationItems />
    </nav>
  </header>
);

export default Toolbar;
