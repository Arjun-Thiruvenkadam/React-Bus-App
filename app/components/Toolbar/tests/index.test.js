import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { MemoryRouter } from 'react-router';
import { IntlProvider } from 'react-intl';
import Toolbar from '../index';
import configureStore from '../../../configureStore';

describe('<Toolbar />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <MemoryRouter>
            <Toolbar />
          </MemoryRouter>
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
