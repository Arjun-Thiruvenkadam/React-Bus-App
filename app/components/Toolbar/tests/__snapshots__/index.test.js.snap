// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toolbar /> should render and match the snapshot 1`] = `
<header
  class="Toolbar"
>
  <div
    class="Logo"
  >
    <img
      alt="Bus"
      src="IMAGE_MOCK"
    />
    <p>
      <span>
        Aju's Bus
      </span>
    </p>
  </div>
  <nav>
    <ul
      class="NavigationItems"
    >
      <li
        class="NavigationItem"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          <span>
            Home
          </span>
        </a>
      </li>
      <li
        class="NavigationItem"
      >
        <a
          href="/order"
        >
          <span>
            Book Tickets
          </span>
        </a>
      </li>
      <li
        class="NavigationItem"
      >
        <a
          href="/login"
        >
          <span>
            Login
          </span>
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
